### PR TITLE
making currency and value dynamic

### PR DESCRIPTION
**Pull request by colins44**
Couple of changes within this PR

Making several fields into text fields so they can be dynamically set my the journey using the syntax ```{{Event.eventDefinitionKey.fieldName}}```

Adding more documentation to make it easier to use the custom activities

Adding new form fields to support new values

### CODE REVIEW

File: public/ixn/activities/promo-codes/customActivity.js [hunk 1]
```diff
@@ -185,10 +185,13 @@ define([
         {'expires_in': $('#promo-code-expires-in').val()},
         {'promo_type': $('#promo-code-type').val()},
         {'promo_value': $('#promo-code-value').val()},
+        {'promo_value_new_user': $('#promo-code-value-new-customer').val()},
         {'currency': $('#promo-code-currency').val()},
         {'min_spend': $('#promo-code-min-spend').val()},
+        {'min_spend_new_user': $('#promo-code-min-spend').val()},
```
Comment (TareevEugene) [inline public/ixn/activities/promo-codes/customActivity.js:191]: promo-code-min-spend-new-customers ?

File: public/ixn/activities/promo-codes/customActivity.js [hunk 2]
```diff
@@ -185,10 +185,13 @@ define([
         {'expires_in': $('#promo-code-expires-in').val()},
         {'promo_type': $('#promo-code-type').val()},
         {'promo_value': $('#promo-code-value').val()},
+        {'promo_value_new_user': $('#promo-code-value-new-customer').val()},
```
Comment (TareevEugene) [inline public/ixn/activities/promo-codes/customActivity.js:188]: promo-code-value-new-customerS ?

Comment (colins44) [reply public/ixn/activities/promo-codes/customActivity.js:188]: Hi @TareevEugene  i was copying the frontend found here: https://admin.luxuryescapes.com/marketing and the post body found within the network call
```
{
"code_name":"colin-testing-103","description":"tseasdfdsf","promo_value":1,"expires_at":"2020-11-06T18:41:59+11:00","promo_type":"percentage","bin_numbers":[],"warning":"","email_addresses":[],"offer_whitelist":[],"offer_blacklist":[],"brand":"luxuryescapes","department_tag":"consumer_marketing","device_types":[],"code_limit":1,"min_spend":1,"currency":"AUD","promo_value_new_user":1,"min_spend_new_user":1,"limit_per_user":1}```

which has `promo_value_new_user`

Comment (TareevEugene) [reply public/ixn/activities/promo-codes/customActivity.js:188]: I'm ok with the name 'promo_value_new_user', i mean that u trying to get value from $('#promo-code-value-new-customer'), but in view it is 'id="promo-code-value-new-customers"'

Comment (colins44) [reply public/ixn/activities/promo-codes/customActivity.js:188]: @TareevEugene thanks for picking that up

**Pull request merged**